### ISSUE TITLE
PHP7 Incompatibility with in registry/json.php with strings

### ISSUE THREAD

**Issue by Crixu44** [joomla/joomla-cms#7571]
#### Steps to reproduce the issue

Install Joomla! on PHp 5.5. 
Then switch to PHP 7.
Error appears

#### Expected result

Joomla! works

#### Actual result

Fatal error: Cannot use Joomla\String\String as String because 'String' is a special class name in [...]/libraries/vendor/joomla/registry/src/Format/Json.php on line 12

#### System information (as much as possible)

PHP 7 Beta 2 
Joomla! 3.4.3

**Comment by Bakual** [comment]
It's a know issue. Joomla 3.4 isn't compatible with PHP7 yet.
Joomla 3.5 should be compatible. To my knowlege all issues are fixed there.

Can you confirm that @mbabker and @wilsonge ?

**Comment by wilsonge** [comment]
Yes we fixed this in the Joomla 3.5 branch by changing the class name to StringHelper. Although we've left the old wrapper String class in place in case people are using it for b/c, we merged it to the 3.5 branch because JString now inherits the StringHelper class rather than String (as they are all static methods this shouldn't be an issue and on top of that the only difference would occur if people initialised and typehinted the class against a class only introduced in 3.4.0)

TLDR: If you test against the 3.5-branch (or 3.5 nightly build at http://developer.joomla.org/cms-packages/) this should be fixed. Closing the issue :)
